### ISSUE TITLE
Add support for shielded instance configuration in google_dataproc_workflow_template

### ISSUE THREAD

**Issue by fulups** [hashicorp/terraform-provider-google#10565]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Add support for shielded instance configuration in **google_dataproc_workflow_template**.
Currently a configuration is available for the gce cluster ([see gce_cluster_config](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataproc_workflow_template#gce_cluster_config)) but
it doesn't support  shielded_instance_config ([see source code](https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_dataproc_workflow_template.go#L1394)).

This support is already implemented in **google_dataproc_cluster** ([see  gce_cluster_config/shielded_instance_config](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataproc_cluster#shielded_instance_config)):

```tf
cluster_config{
  ....
  gce_cluster_config{
    shielded_instance_config{
      enable_secure_boot = true
      enable_vtpm = true
      enable_integrity_monitoring = true
    }
  }
}
```

### New or Affected Resource(s)

* google_XXXXX

### Potential Terraform Configuration

Same format as the configuration available for **google_dataproc_cluster** 

```tf
resource "google_dataproc_workflow_template" "template" {
  ....
  placement {
    managed_cluster {
      cluster_name = "my-cluster"
      config {
        gce_cluster_config {
          ....
          shielded_instance_config{
            enable_secure_boot = true
            enable_vtpm = true
            enable_integrity_monitoring = true
         }
       }
  }
}
```

### References

* https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.workflowTemplates
* https://cloud.google.com/dataproc/docs/reference/rest/v1/ClusterConfig#GceClusterConfig

**Comment by t-indumathy** [comment]
I have started working on this.

**Comment by SarahFrench** [comment]
This issue was addressed in PR #6481 above, and the feature was released in v4.36.0 - [see docs here](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataproc_workflow_template#nested_shielded_instance_config)
